### PR TITLE
Migrate to uv powered pip-compile workflow.

### LINKED ISSUE #121: [Feature Request]: Dependancy Managment Standarization
### Description

Currently, `SpokaneTech_Py` is using a vanilla `requirements.txt` to manage dependencies. I think the project has grown enough that it's time to migrate to a (slightly) more robust system.

While PR #1 proposed a Poetry based system, I've found that a `pip-compile` biased workflow works better for applications. Plus, it only requires `uv`/`pip-tools` to be installed when updating/adding dependencies; `pip` will still work for installing the dependencies.

To migrate, `requirements.txt` and `requirements/dev.txt` will be removed, and moved into `pyproject.toml` under the (new) `[project.dependencies]` and `[project.optional-dependencies]` tables respectively. Then, generate a lock-file using `uv pip compile pyproject.toml -o requirements.lock` (and optionally, lock the development dependencies too: `uv pip compile pyproject.toml --extra dev -o requirements.dev.lock`)

This will slightly change the developer experience, instead of `pip install -r requirements.txt -r requirements/dev.txt` it'll be `pip install -e .[dev]` (or `pip install -r uv.dev.lock`). Plus, Docker builds will be more reproducible.

Additionally, two more tables will need to be added to `pyproject.toml`: `[build-system]` (I prefer Hatchling), and `[project]` (which we should probably have anyways).

TLDR:
Move to a `uv`/`pip-tools` workflow by:
  - [ ] Move `requirements.txt` and `requirements/dev.txt` into `pyproject.toml`
  - [ ] Generate `requirements.lock` and `requirements.dev.lock` (names are flexible)
  - [ ] Add `[project]` and `[build-system]` tables to `pyproject.toml`
  - [ ] Update getting started docs

---

I migrated one of my personal projects to this system about a month ago and have been happy with it so far ([reference PR here](https://github.com/KGB33/RoboShpee/pull/25)). 

I'd appreciate any suggestions or alternatives.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KGB33**
## Pull Request

Closes #121 

Its off-topic, but while I'm changing `pyproject.toml` I could move the Pytest configruation (`pytest.ini`) into it, or a second PR/Issue.

**Checklist:**
- [x] All tests pass.
- [x] Code follows the project's coding standards.
- [x] Documentation has been updated.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,54 @@
+[project]
+name = "SpokaneTech"
+version = "0.1.0"
+description = "Home of SpokaneTech.org, an online hub for Spokane's tech events and groups."
+maitainers = [
```
Comment (joeriddles) [inline pyproject.toml:5]:
```suggestion
maintainers = [
```

**Review by joeriddles — CHANGES_REQUESTED** [review]
Should we remove `requirements.txt`?

**Comment by KGB33** [comment]
> Should we remove `requirements.txt`?

Yep, thanks for catching that.

**Review by joeriddles — APPROVED** [review]
🚀

**Comment by joeriddles** [comment]
> Its off-topic, but while I'm changing pyproject.toml I could move the Pytest configruation (pytest.ini) into it, or a second PR/Issue.

This sounds great

**Pull request merged**